### PR TITLE
INGK-997 Simplify and fix register range method

### PR DESCRIPTION
**Pull request by MartinALL91**
### Type of change

- Simplify method.
- Fix bug.

### Tests
- [x] Run tests.

### Code formatting and linting

- [x] Use the ruff package to format the code: `ruff format ingenialink tests`.
- [x] Use the ruff package to lint the code: `ruff check ingenialink`.

### Others

- [x] Set fix version field in the Jira issue.

### CODE REVIEW

**Comment by GullinBustin** [comment]
Add some tests to check this bug won’t appear again.

File: tests/test_register.py [hunk 1]
```diff
@@ -154,30 +154,25 @@ def test_register_set_storage():
...
-    dtype = REG_DTYPE.U8
-    register = Register(dtype, access)
-    assert register.range == (dtypes_ranges[dtype]["min"], dtypes_ranges[dtype]["max"])
+def test_register_range(dtype, reg_range, reg_type):
+    register = Register(dtype, REG_ACCESS.RW, reg_range=reg_range)
+
+    assert type(register.range[0]) is reg_type
+    assert type(register.range[1]) is reg_type
+    if None in reg_range:
+        assert register.range == (dtypes_ranges[dtype]["min"], dtypes_ranges[dtype]["max"])
```
Comment (GullinBustin) [inline tests/test_register.py:173]: Why, if exist a None in the range, all the range is ignored and the maximum and the minimum of the register type is set?

**Review by GullinBustin — CHANGES_REQUESTED** [review]

File: tests/test_register.py (hunk 1 above)
Comment (MartinALL91) [reply tests/test_register.py]: Done

File: tests/test_register.py [hunk 2]
```diff
@@ -154,30 +154,23 @@ def test_register_set_storage():
     assert register.storage == storage
 
 
+@pytest.mark.parametrize(
+    "dtype, reg_range, expected_range, reg_type",
+    [
+        (REG_DTYPE.U8, (0, 100), (0, 100), int),
+        (REG_DTYPE.FLOAT, (0.0, 1.0), (0.0, 1.0), float),
```
Comment (polfeliu) [inline tests/test_register.py:161]: I would also test FLOAT, (None, None)

**Review by GullinBustin — APPROVED** [review]
